scripts/html-template-brick.js

```diff
@@ -1,18 +1,61 @@
 /**
- * Simpler brick using data-aem-selector attributes in the
+ * Simpler brick using data-aem-inject attributes in the
  * HTML template to select the content to inject in it.
  */
 export default class HtmlTemplateBrick extends HTMLElement {
-  connectedCallback() {
+  static injectFromCSSSelector(scope, elem) {
+    scope.querySelectorAll('*[data-aem-inject]')?.forEach((selector) => {
+      elem.querySelectorAll(selector.dataset.aemInject).forEach((item) => {
+        const src = item.cloneNode(true);
+
+        // remove the attribute so we don't try to inject it again
+        selector.removeAttribute('data-aem-inject');
+
+        if (src) {
+          // set loaded attribute
+          selector.setAttribute('data-aem-inject-loaded', '');
+          selector.append(src);
+        }
+      });
+    });
+  }
+
+  static injectEachLoopFromCSSSelector(scope, elem) {
+    scope.querySelectorAll('*[data-aem-each]').forEach((each) => {
+      const parent = each.parentNode;
+
+      elem.querySelectorAll(each.dataset.aemEach).forEach((item) => {
+        const clone = each.cloneNode(true);
+
+        // remove the attribute so we don't try to inject it again
+        clone.removeAttribute('data-aem-each');
+
+        each.remove();
+
+        // Inject content from our template's CSS selectors
+        HtmlTemplateBrick.injectFromCSSSelector(clone, item);
+
+        // set loaded attribute
+        clone.setAttribute('data-aem-each-loaded', '');
+
+        parent.append(clone);
+      });
+    });
+  }
```

Comment:
As I mentioned in another comment, the queries in each loop iteration are done in the scope of the HTML fragment of the item but this could use some benchmarking and optimizations.

Reply:
Its fine for now